packages/astro/src/core/base-pipeline.ts

```diff
@@ -48,9 +48,13 @@ export abstract class Pipeline {
 		 */
 		readonly site = manifest.site ? new URL(manifest.site) : undefined
 	) {
-		this.internalMiddleware = [
-			createI18nMiddleware(i18n, manifest.base, manifest.trailingSlash, manifest.buildFormat),
-		];
+		this.internalMiddleware = [];
+		// We do use our middleware only if the user isn't using the manual setup
+		if (i18n?.strategy !== 'manual') {
```

Comment:
We also have a similar guard within `createI18nMiddleware`. Does it make sense to colocate the two?